### PR TITLE
Add ability to delete runs in CLI/web

### LINKED ISSUE #26: Allow deletion of profile runs via web-interface
It would be nice to have a button to delete profile runs via the web-interface.

### PR DESCRIPTION
**Pull request by mattboehm**
This allows a run (and all traces with that run_id) to be deleted by:

* running `smiley delete <run_id>`
* clicking the "X" link next to that run in the web view

fixes #26

### CODE REVIEW

**Comment by mattboehm** [comment]
Travis is currently failing due to flake8 warnings for code that was already in place. I'm happy to format that code to shut flake8 up, but frankly think it looks better now than it would with that formatting.

**Comment by dhellmann** [comment]
I'm looking at the flake8 issue in #37

File: smiley/commands/delete.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            'run_id',
+            help='identifier for the run',
+        )
+        return parser
+
+    def take_action(self, parsed_args):
+        db_ = db.DB(parsed_args.database)
+        try:
+            db_.delete_run(parsed_args.run_id)
+            print(u"Deleted run {}".format(parsed_args.run_id))
```
Comment (dhellmann) [inline smiley/commands/delete.py]: The idea behind each command having its own logger object is that the verbosity of the output to the console is controlled through the logging module. So this should be self.log.info(...) to allow the verbose and quiet command line flags to control whether it's actually printed.

File: smiley/commands/delete.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        except db.NoSuchRun:
+            print(u"No run found with id '{}' delete failed".format(
```
Comment (dhellmann) [inline smiley/commands/delete.py]: This could be self.log.warn() since the exception doesn't cause the app to exit with an error.

**Comment by dhellmann** [comment]
This generally looks good, thanks! I have a couple of nits inline. If you add another patch to the PR to fix those up, I can merge it when I get the flake8 tests working correctly again.

**Comment by dhellmann** [comment]
I fixed up the flake8 issues and restarted the travis build for this PR.

**Comment by dhellmann** [comment]
I didn't notice that this doesn't update the documentation. Can you do that, too? docs/source/command_ref.rst

**Comment by mattboehm** [comment]
Yep, I've pushed it. If I understand correctly, I have to open this as a separate pull request since this has already been merged, so I'll do that now.

**Pull request merged**